### ISSUE TITLE
Releases shown against deployments which dont use them.

### ISSUE THREAD

**Issue by tinygrasshopper** [cloudfoundry/bosh#1651]
We are using the [bosh addons](https://bosh.io/docs/runtime-config.html#addons) with the includes functionality to co-locate jobs on specific deployments. But now that we have enabled the addons, we see all other deployments in the output for `bosh deployments` on the director have that release, even though they are not using it. 

Our runtime config:
```
addons:
- name: baz
  include:
    deployments:
    - cf-873319542c61c52c359c
    jobs:
    - name: blobstore
      release: capi
  jobs:
  - name: foo
    release: bar

```

But for other deployments like rabbit, we can see, even the release `bar` isn't used in that deployment: 

```
+-------------------------------------------------------+-----------------------------------+------------------------------------------------+--------------+
| p-rabbitmq-608d1d192db576e210b8                       | bar/0.0.0-rc.88                   | bosh-google-kvm-ubuntu-trusty-go_agent/3363.14 | latest       |
|                                                       | cf-rabbitmq/225.0.0+dev.1         |                                                |              |
|                                                       | consul/152                        |                                                |              |
|                                                       | loggregator/65                    |                                                |              |
|                                                       | on-demand-service-broker/0.15.1   |                                                |              |
|                                                       | rabbitmq-metrics/2.0.0            |                                                |              |
|                                                       | rabbitmq-on-demand-adapter/0.1.4  |                                                |              |
|                                                       | routing/0.143.0                   |                                                |              |
|                                                       | service-metrics/1.5.5             |                                                |              |
+-------------------------------------------------------+-----------------------------------+------------------------------------------------+--------------+
```

cc: @henryaj

**Comment by dpb587-pivotal** [comment]
Agreed this is unfortunate behavior. Looks like there's even a latent todo around unreferenced releases in [planner_factory.rb](https://github.com/cloudfoundry/bosh/blob/993568e93cc40cc6ad1e318e0b129e84cfbcb50e/src/bosh-director/lib/bosh/director/deployment_plan/planner_factory.rb#L93).

**Comment by voelzmo** [comment]
also hit this today. pretty unfortunate :(

**Comment by voelzmo** [comment]
Should be fixed with 266.2.0 and 262.8.0 according to https://www.pivotaltracker.com/n/projects/2132440/stories/155257235

**Comment by voelzmo** [comment]
Still open though: problems with the diff for global addons which aren't used on a specific deployment: https://github.com/cloudfoundry/bosh-cli/issues/360
